src/ConcreteOperations/isdisjoint.jl

```diff
@@ -1178,6 +1166,53 @@ function is_intersection_empty(cup1::UnionSetArray{N},
     return witness ? (result, w) : result
 end
 
+# disambiguation
+function is_intersection_empty(cup::UnionSet{N}, X::AbstractPolyhedron{N},
+                               witness::Bool=false) where {N<:Real}
+    cup_arr = UnionSetArray([cup.X, cup.Y])
+    return _is_intersection_empty_union(cup_arr, X, witness)
+end
+
+function is_intersection_empty(cup::UnionSetArray{N}, X::AbstractPolyhedron{N},
+                               witness::Bool=false) where {N<:Real}
+    return _is_intersection_empty_union(cup, X, witness)
+end
+
+# symmetric disambiguation
+function is_intersection_empty(X::AbstractPolyhedron{N}, cup::UnionSet{N},
+                               witness::Bool=false) where {N<:Real}
+    return is_intersection_empty(cup, X, witness)
+end
+
+function is_intersection_empty(X::AbstractPolyhedron{N}, cup::UnionSetArray{N},
+                               witness::Bool=false) where {N<:Real}
+    return is_intersection_empty(cup, X, witness)
+end
+
+# disambiguation
+function is_intersection_empty(cup::UnionSet{N}, X::AbstractSingleton{N},
+                               witness::Bool=false) where {N<:Real}
+    cup_array = UnionSetArray([cup.X, cup.Y])
+    return _is_intersection_empty_union(cup_array, X, witness)
+end
+
+# disambiguation
+function is_intersection_empty(cup::UnionSetArray{N}, X::AbstractSingleton{N},
+                               witness::Bool=false) where {N<:Real}
+    return _is_intersection_empty_union(cup, X, witness)
+end
+
+# symmetric disambiguation
+function is_intersection_empty(X::AbstractSingleton{N}, cup::UnionSet{N},
+                               witness::Bool=false) where {N<:Real}
+    return is_intersection_empty(cup, X, witness)
+end
+
+# symmetric disambiguation
+function is_intersection_empty(X::AbstractSingleton{N}, cup::UnionSetArray{N},
+                               witness::Bool=false) where {N<:Real}
+    return is_intersection_empty(cup, X, witness)
```

Comment:
The disambiguation for singletons should call the more efficient method that uses membership:

https://github.com/JuliaReach/LazySets.jl/blob/4c5e22586dadfdb2eadb2120cbe22323acc7cc9a/src/ConcreteOperations/isdisjoint.jl#L172-L181